### PR TITLE
PHP 8.1: Explicit octal integer literal notation #5026

### PR DESCRIPTION
**Pull request by junichi11**
- https://github.com/apache/netbeans/issues/5026
- https://wiki.php.net/rfc/explicit_octal_notation
- Fix lexers
- Fix `PHP81UnhandledError`
- Add unit tests for the parser and the lexer

#### PHP 8.1:

nb-php81-explicit-octal-integer-literal-notation

#### PHP 8.0:

nb-php81-explicit-octal-integer-literal-notation-with-php80

### CODE REVIEW

**Comment by junichi11** [comment]
@tmysik Could you please have a look at this? If there is no problem, let's merge it. Thank you!

File: php/php.editor/tools/ASTPHP5Scanner.flex [hunk 1]
```diff
@@ -943,6 +944,12 @@ NOWDOC_CHARS=({NEWLINE}*(([^a-zA-Z_\x7f-\xff\n\r][^\n\r]*)|({LABEL}[^a-zA-Z0-9_\
     pushState(ST_IN_SCRIPTING);
 }
 
+<ST_IN_SCRIPTING>{ONUM} {
+    // PHP 8.1: Explicit octal integer literal notation
+    // https://wiki.php.net/rfc/explicit_octal_notation
+    return createFullSymbol(ASTPHP5Symbols.T_LNUMBER);
```
Comment (tmysik) [inline php/php.editor/tools/ASTPHP5Scanner.flex:950]: Just noticed - for `HNUM` and `BNUM` below, we return `T_DNUMBER` - is that correct?

**Review by tmysik — APPROVED** [review]

File: php/php.editor/tools/ASTPHP5Scanner.flex (hunk 1 above)
Comment (junichi11) [reply php/php.editor/tools/ASTPHP5Scanner.flex:950]: It should not be correct. Actually, `T_LNUMBER` or `T_DNUMBER` is returned in PHP. (`ONUM` too) So need more checks. However, I don't think there are big effects on the existing features of NB currently.

Comment (junichi11) [reply php/php.editor/tools/ASTPHP5Scanner.flex:950]: e.g. 
```php
$hex1 = 0x45fffabcde0000000;
var_dump($hex1); // double(8.0704412748217E+19)

$hex2 = 0x1f;
var_dump($hex2); // int(31)
```

Comment (tmysik) [reply php/php.editor/tools/ASTPHP5Scanner.flex:950]: OK, thanks.

Comment (junichi11) [reply php/php.editor/tools/ASTPHP5Scanner.flex:950]: I'll fix it if we come across a problem :)

Comment (tmysik) [reply php/php.editor/tools/ASTPHP5Scanner.flex:950]: Sure, np.

**Pull request merged**